lightning/src/ln/chan_utils.rs

```diff
@@ -41,6 +41,11 @@ use chain;
 
 const HTLC_OUTPUT_IN_COMMITMENT_SIZE: usize = 1 + 8 + 4 + 32 + 5;
 
+pub(crate) const MAX_HTLCS: u16 = 483;
+
+// This checks that the buffer size is greater than the maximum possible size for serialized HTLCS
+const _EXCESS_BUFFER_SIZE: usize = MAX_BUF_SIZE - MAX_HTLCS as usize * HTLC_OUTPUT_IN_COMMITMENT_SIZE;
```

Comment:
Based on recommendation from @TheBlueMatt - this unsigned subtraction is a compile time check that `MAX_BUF_SIZE > MAX_HTLCS as usize * HTLC_OUTPUT_IN_COMMITMENT_SIZE`.  I believe that this is correct, because it predicts the number of bytes written by `HTLCOutputInCommitment::write`, not the in-memory size.